src/test/java/com/vmware/vipclient/i18n/messages/service/OfflineModeTest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n.messages.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Locale;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.vmware.vip.i18n.BaseTestClass;
+import com.vmware.vipclient.i18n.I18nFactory;
+import com.vmware.vipclient.i18n.VIPCfg;
+import com.vmware.vipclient.i18n.base.cache.Cache;
+import com.vmware.vipclient.i18n.base.cache.MessageCache;
+import com.vmware.vipclient.i18n.base.cache.MessageCacheItem;
+import com.vmware.vipclient.i18n.base.cache.TranslationCacheManager;
+import com.vmware.vipclient.i18n.base.instances.TranslationMessage;
+import com.vmware.vipclient.i18n.exceptions.VIPClientInitException;
+import com.vmware.vipclient.i18n.messages.dto.MessagesDTO;
+
+public class OfflineModeTest extends BaseTestClass {
+
+	String component = "JAVA";
+    String key = "LeadTest";
+    String source = "[{0}] Test alert";
+    Locale locale = new Locale("fil");
+    String comment = "comment";
+    String messageFil = "[{0}] Alerto sa pagsusuri";
+    String messageFr ="[{0}] Alerte de test";
+    Object[] args = { "a" };
+
+    MessagesDTO dto = new MessagesDTO();
+    
+    @Before
+    public void init() {
+        dto.setComponent(component);
+        dto.setKey(key);
+        dto.setSource(source);
+        dto.setLocale(locale.toLanguageTag());
+        VIPCfg.resetInstance();
+    }
+    
+    @Test
+    public void testGetMsgsOfflineMode() {
+    	VIPCfg cfg = VIPCfg.getInstance();
+  
+        try {
+            cfg.initialize("vipconfig-offline");
+        } catch (VIPClientInitException e) {
+            logger.error(e.getMessage());
+        }
+    	
+        Cache c = cfg.createTranslationCache(MessageCache.class);
+        TranslationCacheManager.cleanCache(c);
+        I18nFactory i18n = I18nFactory.getInstance(cfg);
+        TranslationMessage translation = (TranslationMessage) i18n.getMessageInstance(TranslationMessage.class);
+        
+        dto.setProductID(VIPCfg.getInstance().getProductName());
+        dto.setVersion(VIPCfg.getInstance().getVersion());
+        
+    	CacheService cs = new CacheService(dto);
+    	
+    	translation.getString(locale, component, key, source, comment, args);
+    	
+    	MessageCacheItem cacheItem = cs.getCacheOfComponent();
+    	assertNotNull(cacheItem);
+    	assertEquals(messageFil, cacheItem.cachedData.get(key));	
```

Comment:
To get the string, the logic is not coherent, e.g. translation.getString is parallel with cacheItem.cachedData.get(key)
do you think it make sense?
My suggestion is CacheService should be in translation.getString, it should be not exposed to API caller(developer).

Reply:
Are both(transaltion.getString and cacheItem.cachedData.get(key) sequential? What transaltion.getString will return for the first call?